torchao/prototype/safetensors/safetensors_support.py

```diff
@@ -39,15 +45,22 @@ def load_tensor_state_dict(file_path: str, device: str):
     Returns:
         Dictionary of reconstructed tensor subclasses
     """
-    loaded_tensors = load_file(file_path, device)
-
-    with open(file_path, "rb") as f:
-        import struct
-
-        header_size = struct.unpack("<Q", f.read(8))[0]
-        header_bytes = f.read(header_size)
-        header = json.loads(header_bytes)
-        metadata = header.get("__metadata__", {})
+    if do_load_file:
+        assert file_path and device, "Must provide file path to load checkpoint from"
+        loaded_tensors = load_file(file_path, device)
+        with open(file_path, "rb") as f:
+            import struct
+
+            header_size = struct.unpack("<Q", f.read(8))[0]
+            header_bytes = f.read(header_size)
+            header = json.loads(header_bytes)
+            metadata = header.get("__metadata__", {})
```

Comment:
can move this to a util function and move the util function to test_safetensors_support.py I think